src/renderer/shared/api/xcm/xcmService.ts

```diff
@@ -67,35 +65,59 @@ export const getEstimatedWeight = (
 };
 
 export const estimateFee = (
-  instructions: Instructions,
-  baseWeights: NetworkBaseWeight,
+  config: XcmConfig,
   assetLocation: AssetLocation,
   originChain: string,
   xcmTransfer: XcmTransfer,
 ): BN => {
   const weight = getEstimatedWeight(
-    instructions,
+    config.instructions,
     xcmTransfer.destination.fee.instructions,
     new BN(xcmTransfer.destination.fee.mode.value),
   );
 
-  const fee = weightToFee(weight, new BN(baseWeights[xcmTransfer.destination.chainId]));
+  const fee = weightToFee(weight, new BN(config.networkBaseWeight[xcmTransfer.destination.chainId]));
 
   const isReserveChain = [originChain, xcmTransfer.destination.chainId].includes(assetLocation.chainId);
 
   if (isReserveChain) return fee;
 
   const reserveWeight = getEstimatedWeight(
-    instructions,
+    config.instructions,
     assetLocation.reserveFee.instructions,
     new BN(assetLocation.reserveFee.mode.value),
   );
 
-  const reserveFee = weightToFee(reserveWeight, new BN(baseWeights[assetLocation.chainId]));
+  const reserveFee = weightToFee(reserveWeight, new BN(config.networkBaseWeight[assetLocation.chainId]));
 
   return fee.add(reserveFee);
 };
 
+export const estimateRequiredDestWeight = (
+  config: XcmConfig,
+  assetLocation: AssetLocation,
+  originChain: string,
+  xcmTransfer: XcmTransfer,
+): BN => {
+  const weight = getEstimatedWeight(
+    config.instructions,
+    xcmTransfer.destination.fee.instructions,
+    new BN(xcmTransfer.destination.fee.mode.value),
```

Comment:
`fee.mode.value` is only present if `fee,mode.type == "proportional"`. Otherwise we should request node for the fee
By the way, why do we need fee mode when estmiating weight at all? Weight itself only depends on `baseInstructionWeight` and `instructions`